File: c++/leetcode/1281. Subtract the Product and Sum of Digits of an Integer.cpp
class Solution
{
public:
  int subtractProductAndSum(int n)
  {
    int product = 1;
    int sum = 0;
    const int factor = 10;

    while (n)
    {
      product *= n % factor;
      sum += n % factor;
      n = n / factor;
    }

    return product - sum;
  }
};

File: javascript/leetcode/1281. Subtract the Product and Sum of Digits of an Integer.js
/**
 * @param {number} n
 * @return {number}
 */
var subtractProductAndSum = function (n) {
  let number = "" + n;
  let product = 1;
  let sum = 0;

  for (let i of number) {
    product *= parseInt(i);
    sum += parseInt(i);
  }

  return product - sum;
};
